src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -596,6 +596,7 @@ internal static bool IsAssignableFrom(this IEdmType baseType, IEdmType subtype)
                 case EdmTypeKind.Primitive:
                     return ((IEdmPrimitiveType)baseType).IsAssignableFrom((IEdmPrimitiveType)subtype);
 
+                case EdmTypeKind.Untyped:   // fall through

```

Comment:
>case EdmTypeKind.Untyped:   // fall through [](start = 16, length = 43)

What's the scenario for Un-type inheritance?

Reply:
Good question. I think untyped just wasn't handled, and this was getting called (maybe to check the type within a collection was valid for the collection)?

I'll verify.



---
In reply to: [258638045](https://github.com/OData/odata.net/pull/1204#discussion_r258638045) [](ancestors = 258638045)

Reply:
Yes; this is required for reading an untyped collection to make sure that the instances within the collection can be cast to the untyped type.

---
In reply to: [262169653](https://github.com/OData/odata.net/pull/1204#discussion_r262169653) [](ancestors = 262169653,258638045)